pkg/controller/storagecluster/noobaa_system_reconciler.go

```diff
@@ -125,3 +129,70 @@ func (r *ReconcileStorageCluster) newNooBaaSystem(sc *ocsv1.StorageCluster, reqL
 
 	return nb
 }
+
+// Delete all nooba systems in the namespace
+func (r *ReconcileStorageCluster) deleteNoobaaSystemsAndWait(sc *ocsv1.StorageCluster) error {
+	noobaas := &nbv1.NooBaaList{}
+	listOptions := &client.ListOptions {
+		Namespace: sc.Namespace,
+	}
+	err := r.client.List(context.TODO(), listOptions, noobaas)
+	if err != nil {
+		return err
+	}
+
+	for _, noobaa := range noobaas.Items {
```

Comment:
we're only looking to delete a noobaa object with a owner reference pointing to this specific storagecluster. 

I think it would be a good idea to make sure the `sc` here matches the item in the noobaa.OwnerReferences slice.

Reply:
Done.